open-apparel-registry/open-apparel-registry
# Add initial models

## Overview

Adds a set of models based on the schema of the existing MongoDB with some significant structural changes.

Connects #36 

## Notes

We have diverged from and simplified the previous application schema in several ways in an attempt to reduce complexity and increase clarity and self-documentation.

- The Source and Temp models have been converted to User, Organization,
  FactoryList and FactoryListItem.
  - Source was mixing the concepts of the person interacting with the uploads,
    the larger group they are a part of, and the names of uploaded files. This
    is now separated into User, Organization, and FactoryList.
  - The ambiguously named Temp, which represents a row from and uploaded factory
    list CSV, is now FactoryListItem.
- The Address and Geo models have been incorporated into the FactoryListItem and
  Factory models. The function of the previous Address model can be replaced by
  relational queries and we feel it is more appropriate to store the gecoding
  results on the FactoryListItem against which the geocoder is run.
- Fuzzy address matching results were previously stored as an array of objects
  on the Source model and the Confirm model connected them to other models. The
  new FactoryMatch model takes the place of nested objects for storing potential
  matches between FactoryListItem rows and existing Factory rows.

We created a stub user model as recommended by the Django documentation.

We copied the organization type choices from the application front-end code.

## Testing Instructions

* Run `./scripts/update` and verify that the migration is applied without error.
